mailpoet/lib/Settings/SettingsChangeHandler.php

```diff
@@ -77,4 +90,23 @@ private function createScheduledTask(string $type): ScheduledTaskEntity {
     $task->setStatus(ScheduledTaskEntity::STATUS_SCHEDULED);
     return $task;
   }
+
+  public function updateBridge($settings) {
+    $apiKey = $settings[Mailer::MAILER_CONFIG_SETTING_NAME]['mailpoet_api_key'] ?? null;
+    $premiumKey = $settings['premium']['premium_key'] ?? null;
+    if (!empty($apiKey)) {
+      $apiKeyState = $this->bridge->checkMSSKey($apiKey);
+      $this->bridge->storeMSSKeyAndState($apiKey, $apiKeyState);
+    }
+    if (!empty($premiumKey)) {
+      $premiumState = $this->bridge->checkPremiumKey($premiumKey);
+      $this->bridge->storePremiumKeyAndState($premiumKey, $premiumState);
+    }
+    if ($apiKey && !empty($apiKeyState) && in_array($apiKeyState['state'], [Bridge::KEY_VALID, Bridge::KEY_VALID_UNDERPRIVILEGED], true)) {
+      return $this->subscribersCountReporter->report($apiKey);
+    }
+    if ($premiumKey && !empty($premiumState) && in_array($premiumState['state'], [Bridge::KEY_VALID, Bridge::KEY_VALID_UNDERPRIVILEGED], true)) {
+      return $this->subscribersCountReporter->report($apiKey);
```

Comment:
Aha. Good catch! In reality, it has no impact because we use one key, but on the backend, we "pretend" there are be two different keys, but I'll update it.